sdk/core/azure-core/src/main/java/com/azure/android/core/http/interceptor/RetryInterceptor.java

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.core.http.interceptor;
+
+import androidx.annotation.NonNull;
+
+import com.azure.android.core.internal.util.ExceptionUtils;
+import com.azure.android.core.util.DateTimeRfc1123;
+
+import org.threeten.bp.Duration;
+import org.threeten.bp.OffsetDateTime;
+import org.threeten.bp.temporal.ChronoUnit;
+
+import java.io.IOException;
+
+import okhttp3.Interceptor;
+import okhttp3.Request;
+import okhttp3.Response;
+
+/**
+ * Pipeline interceptor that retries when a recoverable exception or HTTP error occurs.
+ */
+public class RetryInterceptor implements Interceptor {
+    private final RetryStrategy retryStrategy;
+
+    /**
+     * Creates {@link RetryInterceptor} with the provided {@link RetryStrategy}.
+     *
+     * @param retryStrategy The {@link RetryStrategy} used for retries.
+     */
+    public RetryInterceptor(RetryStrategy retryStrategy) {
+        this.retryStrategy = retryStrategy;
+    }
+
+    /**
+     * Get an instance of {@link RetryInterceptor} that uses fixed backoff delay retry strategy.
+     *
+     * @param maxRetries The maximum number of times to retry.
+     * @param delay The fixed backoff delay applied before every retry.
+     * @return The retry interceptor.
+     */
+    public static RetryInterceptor fixedDelay(int maxRetries, Duration delay) {
```

Comment:
Maybe it is obvious to people that you will get an interceptor with a specific type of delay with these static methods, however, I would change the name just to be a bit more explicit about it. Right now someone that only looks at the method name could think you are getting some sort of delay from this, instead of the interceptor itself.

```suggestion
    public static RetryInterceptor withFixedDelay(int maxRetries, Duration delay) {
```

Not sure this is the best name for it, but I hope you get the idea :)

Reply:
thanks, Victor! `with*` is good with me. 

One thing we might want to consider later is, is it better if we use `createWith*`? two reasons:

1. This is a factory method, so `create` is a common prefix.
2. The prefix `with` is a common prefix used to qualify setters in the builder/fluent pattern e.g. [here](https://github.com/Azure/azure-libraries-for-java/blob/5280b755fd364ec266c50e78c7fd7acfee382547/azure-samples/src/main/java/com/microsoft/azure/management/compute/samples/ConvertVirtualMachineToManagedDisks.java#L47).

but we can revisit it later but what you proposed is better than the current naming.

Reply:
I'm also ok with using `createWith`, maybe we can bring this up in our next meeting :)